### PR TITLE
Add a option to show the ship name on the Presence text and add pt-br support

### PR DESCRIPTION
**Pull request by Luctins**
Added related code, and edited the language templates (needs fine tuning).

### CODE REVIEW

**Comment by SayakMukhopadhyay** [comment]
Hi, apologies for looking at this so late. I will quickly take a look at this.

**Comment by Luctins** [comment]
I May have accidentally mixed the 2 PR's that I created here. that's why this the other one has a ship name field too.
If it's a problem, I can try to separate the 2.

**Comment by SayakMukhopadhyay** [comment]
It is fine if you submit a single PR with both the ship name stuff and the new translation. Just close the redundant PR to make things clear.

File: L10n/de.strings [hunk 1]
```diff
@@ -1,44 +1,47 @@
...
 
+/* template for ship name */
+" with " = "Deutschverb"
+
 /* Idle message. [load.py] */
 "Connecting CMDR Interface" = "Verbindet mit KMDT-Schnittstelle";
 
 /* Details of system. [load.py] */
-"In system {system}" = "Im System {system}";
+"In system {system}{ship}" = "Im System {system}{ship}";
```
Comment (SayakMukhopadhyay) [inline L10n/de.strings]: I have seen that the ship name is right next to other strings without any spaces. Can you make sure that there is proper spacing in all places?
Maybe do something like `In system {system} flying {ship}` (only if the length aint too much)

File: L10n/de.strings [hunk 2]
```diff
@@ -1,44 +1,47 @@
...
 /* If Hyperspace jumping. [load.py] */
-"Jumping to system {system}" = "Springt zum System {system}";
+"Jumping to system {system}{ship}" = "Springt zum System {system}{ship}";
 
 /* If Supercruise jumping. [load.py] */
 "Preparing for supercruise" = "Macht sich bereit für Supercruise";
 
 /* When supercruising. [load.py] */
-"Supercruising" = "Im Supercruise";
+"Supercruising{ship}" = "Im Supercruise{ship}";
```
Comment (SayakMukhopadhyay) [inline L10n/de.strings:26]: Same spacing issue. See above comments.

File: load.py [hunk 3]
```diff
@@ -202,39 +206,46 @@ def plugin_stop():
 def journal_entry(cmdr, is_beta, system, station, entry, state):
     global planet
     global landingPad
+    global ship
+    if showShipName:
+        #create ship name text if enabled.
+        shipt = " with " + state["ShipName"]
```
Comment (SayakMukhopadhyay) [inline load.py]: Move the `with` in the translation file since that word will vary from language to language

File: load.py [hunk 4]
```diff
@@ -202,39 +206,46 @@ def plugin_stop():
...
+    else:
+        shipt = ""
+
     if entry['event'] == 'StartUp':
         this.presence_state = _('In system {system}').format(system=system).encode('utf-8')
         if station is None:
-            this.presence_details = _('Flying in normal space').encode('utf-8')
+            this.presence_details = _('Flying in normal space{ship}').format(ship=shipt).encode('utf-8')
```
Comment (SayakMukhopadhyay) [inline load.py]: Again spacing needs to be adjusted to be uniform across the board.

**Review by SayakMukhopadhyay — COMMENTED** [review]
Left some comments regarding how things can be made a bit more uniform across the board. I have not commented in all places where changes are required so please do take a look across the codebase where similar fixes can be applied.

**Comment by Luctins** [comment]
> It is fine if you submit a single PR with both the ship name stuff and the new translation. Just close the redundant PR to make things clear.

OK, will try to fix it later today.
Edit: this week has been a little rushed, and i will probably try to finish this on the weekend

File: L10n/de.strings (hunk 1 above)
Comment (Luctins) [reply L10n/de.strings]: That was a mistake on my part when I wrote the find and replace command, fixing with next commit

File: load.py (hunk 3 above)
Comment (Luctins) [reply load.py]: Ok, will do. 
About the spacing, the `{shipname}` param will be in the end of the string with no spacing (the space is inside it) so when disabled there's no extra space at the end.

**Comment by Luctins** [comment]
I've implemented the changes, but I cannot test it yet, because my dev machine doesn't have EDMC set up. will try to test it later today.

**Comment by Luctins** [comment]
It did not work.

Edit: I know what is is, the translation files all have the ship field, but on the code some of them don't (only the detail fields have the ship parameter)

**Comment by Luctins** [comment]
Tested today, has some ruff edges, but it seems to work.

The main things  would be:
* Unaligned checkbox for the "show ship name"  option
like so: 
image

* that the ship name displays as None right after logging
* to check is "Portuguès Brasileiro" is the correct name for the translation

**Comment by Luctins** [comment]
image

**Comment by Luctins** [comment]
image

**Comment by SayakMukhopadhyay** [comment]
Nice work mate! Let me know when you the PR is ready to me merged.

**Comment by Luctins** [comment]
Fixed some last dangling `{ship}` fields.

About merging, the pt-BR translation is ready, but the ship name text makes the detailed presence a little too wide and sometimes doesn't show (when the system name is very long).
Examples (my friend that had his system in pt-BR tested for me):
image
image
Besides that, sometimes, the `state["shipName"]` field come as `None` and currently it hides the ship name text in that case, I will investigate this and comment here if there's any change.

**Comment by SayakMukhopadhyay** [comment]
Maybe put the ship name in a new line?

**Comment by Luctins** [comment]
Sadly that doesn't work: 
`shipt = " \nwith {ship}".format(ship=state['ShipName'])`
image
It seems like discord prevents you from inserting newlines into the presence text.
Later this week I'll try to remove the ship text code, and push only the translation.
Maybe using the image field I can try to implement this later (in another PR).

**Comment by SayakMukhopadhyay** [comment]
> Later this week I'll try to remove the ship text code, and push only the translation.

Alright!

**Comment by Luctins** [comment]
colsing this and opening one only for the translation.

**Pull request closed without merging**